lib/postgresql.js

```diff
@@ -132,15 +108,19 @@ PostgreSQL.prototype.executeSQL = function (sql, params, callback) {
       self.debug('SQL: ' + sql);
     }
   }
-  self.connection.query(sql, params, function (err, data) {
-    // if(err) console.error(err);
-    if (err && self.settings.debug) {
-      self.debug(err);
-    }
-    if (self.settings.debug && data) self.debug("%j", data);
-    // console.log(err);
-    if (log) log(sql, time);
-    callback(err ? err : null, data ? data.rows : null);
+
+  self.pg.connect(self.clientConfig, function(err, client, done) {
+    client.query(sql, params, function(err, data) {
+      // if(err) console.error(err);
+      if (err && self.settings.debug) {
+        self.debug(err);
+      }
+      if (self.settings.debug && data) self.debug("%j", data);
+      // console.log(err);
+      if (log) log(sql, time);
+      process.nextTick(done); // Release the pooled client
+      callback(err ? err : null, data ? data.rows : null);
+    });
```

Comment:
My concern is that if you have 20 clients served at the same tick, then you need to open 20 connections. While if the connection was returned immediately, there is a chance that the next client can pick up the connection returned by the previous client.

Anyways, this is most likely a theoretical issue at the moment and it can be always easily fixed later.